### PR TITLE
New arch/fab handling

### PR DESCRIPTION
**Pull request by jabarros**
## Related Issues
App:

Library PR (if needed):

- [ ] Added changelog files for the fixed issues in folder changelog/unreleased. More info [here](https://github.com/owncloud/android/tree/master/changelog#create-changelog-items)
_____

### CODE REVIEW

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/filelist/MainFileListFragment.kt [hunk 1]
```diff
@@ -41,20 +49,32 @@ import com.owncloud.android.presentation.ui.files.SortOptionsView
...
 import org.koin.androidx.viewmodel.ext.android.viewModel
+import org.koin.java.KoinJavaComponent.get
 
-class MainFileListFragment : Fragment(), SortDialogListener, SortOptionsView.SortOptionsListener, SortOptionsView.CreateFolderListener {
+class MainFileListFragment : Fragment(), SortDialogListener, SortOptionsView.SortOptionsListener, SortOptionsView.CreateFolderListener,
+    CreateFolderDialogFragment.CreateFolderListener {
 
     private val mainFileListViewModel by viewModel<MainFileListViewModel>()
 
+    val ARG_LIST_FILE_OPTION = MainFileListFragment::class.java.canonicalName +
```
Comment (fesave) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/filelist/MainFileListFragment.kt:64]: I think we should add this argument in the companion object like the other arguments, also, you can use the same structure as others to concatenate the strings. what do you think?

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/filelist/MainFileListFragment.kt [hunk 2]
```diff
@@ -196,7 +227,142 @@ class MainFileListFragment : Fragment(), SortDialogListener, SortOptionsView.Sor
...
+     *
+     * @param enabled Desired visibility for the FAB.
+     */
+    private fun setFabEnabled(enabled: Boolean) {
+        if (enabled) {
+            binding.fabMain.visibility = View.VISIBLE
+        } else {
+            binding.fabMain.visibility = View.GONE
+        }
+    }
```
Comment (fesave) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/filelist/MainFileListFragment.kt:269]: I think that we can reduce this method, like this:

```
    private fun setFabEnabled(enabled: Boolean) 
    {
       binding.fabMain.isVisible = enabled
    }
```
What do you think?

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/filelist/MainFileListFragment.kt [hunk 3]
```diff
@@ -196,7 +227,142 @@ class MainFileListFragment : Fragment(), SortDialogListener, SortOptionsView.Sor
...
+        snackbar.show()
+    }
+
+    private fun showSnackMessage(messageResource: String) {
+        val snackbar = Snackbar.make(
+            requireActivity().findViewById(R.id.coordinator_layout),
+            messageResource,
+            Snackbar.LENGTH_LONG
+        )
+        snackbar.show()
```
Comment (fesave) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/filelist/MainFileListFragment.kt:350]: I see we have two methods to display a SnackMessage, could we make an extension function (Thanks to kotlin :D) where we just pass the message? We could reuse it in any fragment.

**Review by fesave — CHANGES_REQUESTED** [review]
Great job! I leave you a few comments to value, if you have any questions, ask me! :)

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/filelist/MainFileListFragment.kt (hunk 1 above)
Comment (jabarros) [reply owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/filelist/MainFileListFragment.kt]: Sure!

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/filelist/MainFileListFragment.kt (hunk 2 above)
Comment (jabarros) [reply owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/filelist/MainFileListFragment.kt:262]: 👍

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/filelist/MainFileListFragment.kt (hunk 3 above)
Comment (jabarros) [reply owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/filelist/MainFileListFragment.kt]: We have an existing extension. Used!

**Review by fesave — APPROVED** [review]
Perfect! 🏅

**Pull request merged**